mne/epochs.py

```diff
@@ -1810,6 +1810,56 @@ def save(self, fname, split_size='2GB', fmt='single', overwrite=False,
             this_epochs.event_id = self.event_id
             _save_split(this_epochs, fname, part_idx, n_parts, fmt)
 
+    def export(self, fname, fmt="auto"):
+        """Export Epochs to EEGLAB's .set format.
+
+        Parameters
+        ----------
+        fname : str
+            Name of the export file.
+        fmt : str
+            Format of the export. Defaults to auto, which will infer
+            format from fname's extension
+
+        Notes
+        -----
+        Channel locations are expanded to the full EEGLAB format
+        For more details see eeglabio.utils.cart_to_eeglab
+        """
+        if fmt == "auto":
+            fmt = op.splitext(fname)[1]
+            if fmt:
+                fmt = fmt[1:]
+            else:
+                raise ValueError("Couldn't infer format from filename "
+                                 "(no extension found)")
+
+        if fmt == "set":
+            from eeglabio.epochs import export_set
+            # load data first
+            self.load_data()
+
+            # remove extra epoc and STI channels
+            drop_chs = ['epoc', 'STI 014']
+            pick_chs = [ch for ch in self.ch_names if ch not in drop_chs]
+
+            chs = self.info["chs"]
+            cart_coords = np.array([d['loc'][:3] for d in chs
+                                    if d['ch_name'] not in drop_chs])
+            if cart_coords.any():  # has coordinates
+                # (-y x z) to (x y z)
+                cart_coords[:, 0] = -cart_coords[:, 0]  # -y to y
+                # swap x (1) and y (0)
+                cart_coords[:, [0, 1]] = cart_coords[:, [1, 0]]
+            else:
+                cart_coords = None
+
+            export_set(fname, self.get_data(picks=pick_chs),
+                       self.info['sfreq'], self.events, self.tmin, self.tmax,
+                       pick_chs, self.event_id, cart_coords)
+        else:
+            raise ValueError("Format not supported (%s)" % fmt)
```

Comment:
maybe a `NotImplementedError` is more appropriate here?  I can also see the logic of making a distinction between known formats not yet supported (`NotImplementedError`) and unrecognized formats (`ValueError`).

Reply:
I think `NotImplementedError` would be appropriate for other to be implemented cases in `elif` statements, but `ValueError` in `else` will catch the unknown formats. 

Perhaps it's appropriate to have something similar to [`read_custom_montage`](https://github.com/mne-tools/mne-python/blob/176680a238a7c4f132fdd9198fc7bbe7797d764f/mne/channels/montage.py#L1106) considering the BrainVision case you mentioned before.